AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/Persistence/LogActor.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Combine
+import Foundation
+
+/// Wrapper around a LogRotation to ensure
+/// thread-safe usage.
+actor LogActor {
+    
+    private let rotation: LogRotation
+    private let rotationSubject: PassthroughSubject<URL, Never>
+    
+    /// Initialized the actor with the given directory and fileCountLimit.
+    init(directory: URL, fileSizeLimitInBytes: Int) throws {
+        self.rotation = try LogRotation(directory: directory, fileSizeLimitInBytes: fileSizeLimitInBytes)
+        self.rotationSubject = PassthroughSubject()
+    }
+    
+    /// Attempts to persist the given log entry.
+    func record(_ entry: LogEntry) throws {
```

Comment:
I don't think it's an issue here as order aren't important since every log entry has a corresponding timestamp but I'll log a task to revisit and potentially refactor this as a followup after release.